Add Periphers for Heltec with GPS Module Details

5kamisama5 opened the pull request:
I will like to add more instructions for people to hook up their modules to for specific boards. This is an example of one which I tried to be as detailed as possible for newer members, this will keep the same question from being asked on the #help channel. I have another for CardKB and buzzer modules.

5kamisama5 commented:
![Hook up GPS module](https://github.com/meshtastic/meshtastic/assets/116696711/20ee5170-a72a-4b2e-be89-4181e85d39d2)
Here's the picture I want to add

5kamisama5 commented:
@rcarteraz I've made the changes suggested. Thanks for the suggestion. Hopefully it's better

tekstrand commented:
@5kamisama5 you'll need to update the directories/files in this, the naming structure currently uses spaces and does not define a file type. Make sure you match the other files in the docs directory

tekstrand commented:
hey @rcarteraz did you mean to assign this to me?

5kamisama5 commented:
> @5kamisama5 you'll need to update the directories/files in this, the naming structure currently uses spaces and does not define a file type. Make sure you match the other files in the docs directory

Sorry, not exactly sure what mean. i did add .mdx to the change I just did, if that's what you mean.

tekstrand commented:
<img width="487" alt="Screenshot 2024-02-15 at 7 54 12 PM" src="https://github.com/meshtastic/meshtastic/assets/5215365/eae6e16d-42a8-4fd5-a163-708424e1ad80">
See how your file names have spaces in them? They should match the other files in this repo. Same with the image. You can't use spaces in file names. You'll want to organize the new files into existing folders

tekstrand commented:
@pdxlocations @rcarteraz should this be under hardware, or is this more of a project? I'm really not seeing any other content like this under `hardware`

pdxlocations commented:
> @pdxlocations @rcarteraz should this be under hardware, or is this more of a project? I'm really not seeing any other content like this under `hardware`

I think hardware is where it is makes sense since it's specific to a Heltec peripheral (as far as pinouts). maybe heltec/peripherals/ similar to RAK https://meshtastic.org/docs/hardware/devices/rak/peripherals/

Not 100% sure of that statement above, but it would be good to come up with a standard for this type of integration.

tekstrand commented:
I think peripherals makes sense 👍

5kamisama5 commented:
Formatting is now similar to the RAK one.

pdxlocations commented:
Could someone deploy a preview of this?  I'm not on the Vercel team.

tekstrand commented:
Seems to be a `tab` html tag left open. I do think we need to get rid of the extra folder and just put this content under a `peripherals.mdx` though, to match how raspi does it.

5kamisama5 commented:
> Seems to be a `tab` html tag left open. I do think we need to get rid of the extra folder and just put this content under a `peripherals.mdx` though, to match how raspi does it.

Updated to follow the directory of the Raspberry but I think people may think this applies to the other heltec devices. I don't know how to make it specific without my original directory that specified the V3. I also corrected the missing tab

tekstrand commented:
Couple of things left
1. Need to update the filename on the image to not have spaces
2. I don't think the image is actually used in your document, if you want to have it visible you'll need to add it. [Docusaurus has some docs](https://docusaurus.io/docs/markdown-features/assets#images) on how to do that if you aren't familiar

5kamisama5 commented:
I removed the spaces from the image name, and added the image to the document. Under wiring diagram @tekstrand 

![Alt text for the image](static/img/hardware/HeltecESP32V3-gps-module.webp)

tekstrand commented:
The build is now failing because of a missing closing tag. If you click on `Details` to the right of `CI / build (pull request) you can see the details
```
(node:1927) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[INFO] [en] Creating an optimized production build...
[info] [webpackbar] Compiling Client
[info] [webpackbar] Compiling Server
[success] [webpackbar] Client: Compiled with some errors in 1.14m
Error:  Client bundle compiled with errors therefore further build is impossible.
Error: MDX compilation failed for file "/home/runner/work/meshtastic/meshtastic/docs/hardware/devices/heltec/peripherals.mdx"
Cause: Expected a closing tag for `</TabItem>` (11:1-19:4)
Details:
{
  "message": "Expected a closing tag for `</TabItem>` (11:1-19:4)",
  "name": "1:1",
  "reason": "Expected a closing tag for `</TabItem>` (11:1-19:4)",
  "ruleId": "end-tag-mismatch",
  "source": "mdast-util-mdx-jsx"
}
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

I think it's just missing the closing `</Tabs>` maybe?

tekstrand commented:
Alternatively if you're running the project locally you can test the build on your machine by running `pnpm build`

5kamisama5 commented:
finally found the missing </Tabs>

tekstrand commented on docs/hardware/devices/heltec/peripherals.mdx:
```diff
@@ -0,0 +1,90 @@
+ 9. Set GPS_TX_PIN to 47 (This will communicate to the RXD slot on the GPS)
+ 10. Set PIN_GPS_EN to 46 (This will allow the meshtastic firmware to turn off the power on the GPS board with the user button of the Heltec Board by pressing it 3 times)
+
+## Wiring Diagram
```
Sorry, just noticed one last thing, this, and the headings below it, should all be >3 as they are all children of GPS Module which is a 2

tekstrand commented on docs/hardware/devices/heltec/peripherals.mdx:
```diff
@@ -0,0 +1,90 @@
+
+## Keyboard
+
+Commming Soon
```
Extra m here and under buzzer

5kamisama5 commented:
Thanks for the correction, this was such a learning experience. @tekstrand

tekstrand reviewed (approved):
Great job @5kamisama5 ! Thank you for sticking with it and making all of these changes. This will definitely enhance the documentation.
